.github/workflows/changelogs/changelogs.py

```diff
@@ -0,0 +1,639 @@
+        self.file = file
+        # TODO: Pinpoint the column number for issues where applicable
+        self.line = line
+        self.end_line = end_line
+        self.package = package
+
+    def get_message_header(self) -> str:
+        """Return a message prefix for the issue
+
+        The prefix depends on the issue type, the environment, and the
+        additional information available.
+        """
+
+        # Prepend special workflow tags if running as a GitHub action
+        if os.getenv("GITHUB_ACTION"):
+            msg = "::error" if self.severe else "::warning"
+            if self.file:
+                msg += " file={}".format(self.file)
+                if self.line:
+                    msg += ",line={}".format(self.line)
+                    if self.end_line:
+                        msg += ",endLine={}".format(self.end_line)
+
+            return msg + "::"
+        else:
+            return "ERROR: " if self.severe else "WARNING: "
+
+    def __str__(self):
+        out = self.get_message_header()
+        out += self.msg
+
+        if self.package:
+            out += " for package " + self.package
+        if self.file:
+            out += " in file " + self.file
+            if self.line:
+                out += "#L" + str(self.line)
+                if self.end_line:
+                    out += "-" + str(self.end_line)
+        return out
+
+class ChangelogValidator:
+    """Class that handles the changelog validation
+
+    Validates the changelog entries by enforcing various rules, such as;
+      - Basic typos, spacing, capitalization and line length issues
+      - Ensuring the changelogs are added for correct packages
+
+    Additionally, the following checks are made if a tracker file is
+    provided;
+      - Correct spelling of tracker IDs defined by IBS/OBS
+      - Correct matching of tracker IDs between the PR title, comment
+        messages and the changelog entries (only if a valid PR number is
+        provided)
+      - For Bugzilla trackers, the bug exists in the specified Bugzilla
+        host
+      - For Bugzilla trackers, the bug is reported for the "SUSE Manager"
+        product (passes with a warning if the check fails)
+
+    If a tracker file and a PR number is provided, the following environment
+    variables must be also set:
+
+        GH_TOKEN: A GitHub access token with the basic privileges
+        BZ_TOKEN: A valid Bugzilla API key
+
+    The trackers XML file (trackers.xml):
+        This is an XML document that contains definitions and formats of all
+        the various types of tracker IDs recognized by IBS/OBS. The document is
+        provided by the OBS/IBS APIs.
+
+        It can be obtained by sending a request to the '/issue-trackers'
+        endpoint of the OBS/IBS API:
+
+        `osc api /issue-trackers`
+    """
+
+    def __init__(self, spacewalk_root: str, git_repo: str, pr_number: int, max_line_length: int,
+                 regex_rules: type[RegexRules]):
```

Comment:
Afaik, `type[RegexRules]` could be `RegexRules`. Did something not work without using `type[]`?

Reply:
Looks like you're correct. I wonder why it didn't work the first time..